sljit_src/sljitNativeLOONGARCH_64.c

```diff
@@ -3652,16 +3681,45 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_s32 sljit_emit_atomic_store(struct sljit_compiler
 	CHECK_ERROR();
 	CHECK(check_sljit_emit_atomic_store(compiler, op, src_reg, mem_reg, temp_reg));
 
+	if (!(LOONGARCH_CFG2_LAMCAS & get_cpu_features(GET_CFG2))) {
+		switch (GET_OPCODE(op)) {
+		case SLJIT_MOV:
+		case SLJIT_MOV_P:
+			ins = SC_D;
+			break;
+		case SLJIT_MOV_S32:
```

Comment:
SLJIT_MOV_U32?

signed types should return SLJIT_ERR_UNSUPPORTED AFAIK

Reply:
RISC-V and MIPS all use almost the same ll/sc for atomic load/store as LoongArch, and they all support signed types, so I think it wouldn't be an issue. BTW, I have noticed that in sljit_src/sljitNativeMIPS_common.c line 4234 has a SLJIT_CONFIG_RISCV_64 comment, should be a typo.  @zherczeg

Reply:
> signed types should return SLJIT_ERR_UNSUPPORTED AFAIK

I stand corrected, apparently the "spec" was changed with 8bcd711708fd429eeefe24031c6a5f4decb2e4c5, so guess that the ones still rejecting signed types might be the ones that are not up to spec.

Eitherway you are still missing SLJIT_MOV_U32 there, right?

Reply:
Is it okay to use signed 32-bit ll/sc for 32-bit unsigned atomic load/store directly? I'm worry about boundary cases.

Reply:
definitely not but is better than leaving it out as unsupported, and indeed I was surprised when it didn't break (as it did in a previous attempts to implement this as shown by #172).

which is why I was curious about the bug fixed, since it seems obvious that our test coverage needs some serious overhaul.